project_1/part_2.py

```diff
@@ -0,0 +1,152 @@
+import numpy as np
+import cv2
+import pdb
+from matplotlib import pyplot as plt
+import pdb
+
+class project:
+
+    def img_crop(self, c_img):
+        '''
+        Crops image around the center
+        Args: cv2 image object
+        Return: cropped cv2 image object
+        '''
+        rows, columns, channel = c_img.shape
+        nrows = rows/2
+        ncolumns = columns/2
+        return(c_img[(nrows - 500):(nrows + 500), (ncolumns - 500):(ncolumns + 500)])
+
+    def img_avg(self, a_img, color_channel):
+        '''
+        Averages the brightness of the image
+        Args: cv2 image object
+        Return: averaged float value
+        '''
+        rows, columns, channel = a_img.shape
+        total_sum = 0.0
+        # 0 - blue, 1 - green, 2 - red
+        for i in range(rows):
+            for j in range(columns):
+                total_sum = total_sum + a_img[i, j, color_channel]
+        return(total_sum/(a_img.size))
+
+    def part_1_init(self, color_channel):
+        '''
+        Initializes images and computes average brightness
+        Args: None
+        Return: Brightness (B), Exposure time (T), log B, logT, mean B, mean T
+        '''
+        img1 = cv2.imread('IMG_1_350.JPG')
+        img2 = cv2.imread('IMG_2_250.JPG')
+        img3 = cv2.imread('IMG_3_180.JPG')
+        img4 = cv2.imread('IMG_4_125.JPG')
+        img5 = cv2.imread('IMG_5_90.JPG')
+        img6 = cv2.imread('IMG_6_60.JPG')
+        img7 = cv2.imread('IMG_7_45.JPG')
+        img8 = cv2.imread('IMG_8_30.JPG')
+
+        # image 1
+        nimg1 = self.img_crop(img1)
+        aimg1 = self.img_avg(nimg1, color_channel)
```

Comment:
```python
B, T, log_B, log_T, mean_log_B, mean_log_T = self.part_1_init(color_channel)

# linear regression on log_B
linearized_log_B, g_inv = self.linear_regression(log_T, log_B, mean_log_T, mean_log_B)
```

I just took out the self in both these places and it works fine. I think the problem is you have self in all the calls which is giving you problems.